src/features/auth/views/Signup.tsx

```diff
@@ -0,0 +1,153 @@
+import { Box, Grid, Link, Typography } from "@mui/material"
+import styled, { css } from "styled-components"
+import { Form, Formik } from "formik"
+import Button from "@/components/Button/Button"
+import FormikTextField from "@/components/FormikTextField/FormikTextField"
+import registerSchema, {
+  RegisterFormValues,
+} from "@/libs/yup/schemas/registerSchema"
+import { Link as RouterLink, useNavigate } from "react-router-dom"
+import { useAppDispatch, useAppSelector } from "@/app/hooks"
+import { registerUser, setIsRegistrationCompleted } from "../authSlice"
+import { useTranslation } from "react-i18next"
+
+const Signup = () => {
+  const { t } = useTranslation(["forms", "common"])
+  const dispatch = useAppDispatch()
+  const navigate = useNavigate()
+  const { isRegistrationCompleted } = useAppSelector((store) => store.auth)
+  const initialValues: RegisterFormValues = {
+    firstName: "",
+    lastName: "",
+    email: "",
+    password: "",
+    confirmPassword: "",
+  }
+
+  if (isRegistrationCompleted) {
+    navigate("/login")
+    dispatch(setIsRegistrationCompleted(false))
+  }
+
+  return (
+    <StyledSignup>
+      <Box p={2}>
+        <Box
+          textAlign="center"
+          display="flex"
+          flexDirection="column"
+          gap={1}
+          my={1}
+        >
+          <Typography variant="h4">{t("common:signup")}</Typography>
+          <Typography>{t("forms:quickEasy")}</Typography>
+        </Box>
+        <Formik
+          initialValues={initialValues}
+          onSubmit={async ({ firstName, lastName, email, password }) => {
+            await dispatch(
+              registerUser({ firstName, lastName, email, password }),
+            )
+          }}
+          validationSchema={registerSchema}
+        >
+          {({ isSubmitting }) => (
+            <Form>
```

Comment:
OPTIONAL: You can migrate this form inside auth/components/RegisterForm/RegisterForm.tsx to make your Login component thinner